File: hj/TwoIntegerSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> um;

        for (int i = 0; i < nums.size(); i++) {
            int gap = target - nums[i];
            if (um.find(gap) != um.end()) {
                return i < um[gap] ?
                    vector<int>{i, um[gap]} : vector<int>{um[gap], i};
            }
            um[nums[i]] = i;
        }
        return {};
    }
};

File: Changwhi/two-integer-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = 0
        r = 1
        while l < len(nums) - 1:
            if nums[l] + nums[r] == target:
                return [l, r]
            elif r >= len(nums) - 1:
                l += 1
                r = l + 1
            else:
                r += 1
                
        return []
        